constructor/preconda.py

```diff
@@ -328,18 +328,25 @@ def copy_extra_files(
     Returns:
         list[os.PathLike]: List of normalized paths of copied locations.
     """
+
+    def check_file_exists(file_path):
```

Comment:
```suggestion
    def check_file_exists(file_path: str) -> Path:
```

Let's try and add typing where we can. I'm not sure about that function name either - checking isn't really its primary purpose (or that would return a `bool`).